lib/Dialect/Emit/EmitOps.cpp

```diff
@@ -0,0 +1,70 @@
+//===- EmitOps.cpp - Implement the Emit operations ------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This file implements `sim` dialect ops.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/Emit/EmitOps.h"
+
+using namespace mlir;
+using namespace circt;
+using namespace emit;
+
+//===----------------------------------------------------------------------===//
+// FileOp
+//===----------------------------------------------------------------------===//
+
+void FileOp::build(OpBuilder &builder, OperationState &result,
+                   StringRef symName, StringRef fileName,
+                   llvm::function_ref<void()> bodyCtor) {
+  MLIRContext *context = builder.getContext();
+  OpBuilder::InsertionGuard guard(builder);
+
+  result.addAttribute("sym_name", StringAttr::get(context, symName));
```

Comment:
Can this dialect opt into properties instead of attributes since it's new?

Reply:
We can use some properties, but I'll keep the filenames and text as attributes since string property handling seems to be quite rough (nonexistent?) in MLIR.